src/lib/cloud-sdk-api.js

```diff
@@ -191,6 +245,141 @@ class CloudSdkAPI {
       throw `Error: ${change.status} - ${change.statusText}`
     }
   }
+
+  async _requestJson(callback) {
+    let response = await callback();
+    if (response.status === 200) {
+      return await (response.json());
+    } else {
+      throw `Error: ${response.status} - ${response.statusText}`
+    }
+  }
+
+  async _waitForStatus(predicate, request) {
+      let status = await this._requestJson(request);
+
+      while (!predicate(status)) {
+        await sleepSeconds(10);
+        status = await this._requestJson(request);
+      }
+
+      return status;
```

Comment:
Is the suggested change easier to understand? I definitely would rename `status` to something clearer. `json` seems ok to me, because the method is pretty general purpose.

```suggestion
      while (true) {
        let json = await this._requestJson(request);
        if (predicate(json)) {
          return json
        }
        await sleepSeconds(10);
      }
```

Reply:
Renaming status to json makes sense - not sure about changing the loop.

Reply:
In these kind of retry situations, I try to avoid the repetition of the invocation that does the actual work (i.e. `this._requestJson(request)`. However, I haven't found a universally applicable solution that really satisfies me. So it's also ok to just rename to `json` and be done with it.